### PR TITLE
refactor: Cleanup internals of ProductSliderCmsElementResolver

### PR DESCRIPTION
**Pull request by aragon999**
### 1. Why is this change necessary?
While looking at the recent changes, I noticed that the code of the `ProductSliderCmsElementResolver` is quite confusing.

### 2. What does this change do, exactly?
Cleanup the internals of the `ProductSliderCmsElementResolver`.

### 3. Describe each step to reproduce the issue or behaviour.
Look at the code.

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfill them.

### CODE REVIEW

**Comment by marcelbrode** [comment]
@MartinKrzykawski Since you're currently cleaning up there as well, this is probably interesting for you

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php [hunk 1]
```diff
@@ -178,29 +172,24 @@ private function collectByProductStream(ResolverContext $resolverContext, FieldC
...
-        $productStreamLimit = $elementConfig->get('productStreamLimit');
-        if ($productStreamLimit !== null) {
-            $limit = $productStreamLimit->getIntValue();
-        }
-
         $criteria = new Criteria();
         $criteria->addFilter(...$filters);
-        $criteria->setLimit($limit);
+        $criteria->setLimit($elementConfig->get('productStreamLimit')?->getIntValue() ?? self::FALLBACK_LIMIT);
         $criteria->addAssociation('options.group');
```
Comment (ghost) [inline src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php]: $criteria->addAssociations(self::PRODUCT_ASSOCIATIONS);

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php [hunk 2]
```diff
@@ -31,9 +32,16 @@ class ProductSliderCmsElementResolver extends AbstractCmsElementResolver
     private const PRODUCT_SLIDER_ENTITY_FALLBACK = 'product-slider-entity-fallback';
     private const STATIC_SEARCH_KEY = 'product-slider';
     private const FALLBACK_LIMIT = 50;
+    private const PRODUCT_ASSOCIATIONS = [
+        'cover',
```
Comment (ghost) [inline src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php]: 'cover' can be removed.
See https://github.com/shopware/shopware/pull/5260/files/aab1539965768738b97aa03178987ec0420d3ca7#r1814389953

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php [hunk 3]
```diff
@@ -31,9 +32,16 @@ class ProductSliderCmsElementResolver extends AbstractCmsElementResolver
     private const PRODUCT_SLIDER_ENTITY_FALLBACK = 'product-slider-entity-fallback';
     private const STATIC_SEARCH_KEY = 'product-slider';
     private const FALLBACK_LIMIT = 50;
+    private const PRODUCT_ASSOCIATIONS = [
```
Comment (ghost) [inline src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php:39]: 'PRODUCT_ASSOCIATIONS' should be deprecated to get removed with Shopware 6.7
See https://github.com/shopware/shopware/issues/5326 & https://github.com/shopware/shopware/pull/5260#issuecomment-2444159644

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php (hunk 1 above)
Comment (aragon999) [reply src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php]: Thank you :-) I missed that place.

File: src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php (hunk 3 above)
Comment (aragon999) [reply src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php:39]: Ah, perfect, yes that is reasonable :-)

**Review by ghost — APPROVED** [review]

**Comment by MartinKrzykawski** [comment]
Hello @aragon999,

thanks a lot for contribution.

Your changes have been successfully merged!

Best regards

**Comment by DMX4ALLGmbH** [comment]
thanks bro's

**Pull request closed without merging**